webui/react/src/pages/ExperimentDetails/TrialsComparisonModal.tsx

```diff
@@ -82,6 +82,7 @@ export const TrialsComparisonTable: React.FC<TableProps> = ({
   const [trialsDetails, setTrialsDetails] = useState(trials ?? []);
   const [selectedHyperparameters, setSelectedHyperparameters] = useState<string[]>([]);
   const [selectedMetrics, setSelectedMetrics] = useState<Metric[]>([]);
+  const colSpan = Array.isArray(experiment) ? experiment.length + 1 : 1;
```

Comment:
nit: I think this is equivalent to `(experiment.length ?? 0) + 1`

Reply:
experiment type is `ExperimentItem | ExperimentItem[]`, so when the type is `ExperimentItem`, i think we cant use `experiment.length ?? 0` pattern.

Reply:
I believe that since ExperimentItem doesn't have a length property it should return undefined making the statement work, but I'm fine leaving it as is if you think it's unclear.

Reply:
Do you mean this?

```ts
const test = (experiment.length ?? 0) + 1;
// OR
const test2 = (experiment?.length ?? 0) + 1;
```

In that case, I think it doesnt work in typescript due to `Property 'length' does not exist on type 'ExperimentItem | ExperimentItem[]'.
  Property 'length' does not exist on type 'ExperimentItem'.`
I think it would work in JS. 

```ts
const test3 = 'length' in experiment ? experiment.length + 1 : 1;
```

would work but

```ts
const colSpan = Array.isArray(experiment) ? experiment.length + 1 : 1;
```

is more readable.

<img width="956" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/befb1619-a4d6-4aa1-8be3-a247c57c773b">

Reply:
Ah, yeah, typescript would have an issue with that. I had tested it out quickly in the browser console. What you have is fine.